hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/BaseSparkInternalRowReaderContext.java

```diff
@@ -101,6 +123,25 @@ public HoodieRecord<InternalRow> constructHoodieRecord(Option<InternalRow> rowOp
     return new HoodieSparkRecord(row, HoodieInternalRowUtils.getCachedSchema(schema));
   }
 
+  @Override
+  public HoodieRecord<InternalRow> constructHoodieRecord(Option<InternalRow> rowOption,
+                                                         Map<String, Object> metadataMap,
+                                                         TypedProperties props) {
+    if (!rowOption.isPresent()) {
```

Comment:
again, can you help me understand why do we need these adhoc changes ?

Reply:
This is also related to the record key issue.